packages/bcer-data-portal/app/src/hooks/useLeaflet.tsx

```diff
@@ -0,0 +1,338 @@
+  const [markers, setMarkers] = useState<L.Marker[]>([]);
+  const [geoJsonData, setGeoJsonData] = useState<L.GeoJSON>();
+  const [startingLocation, setStartingLocation] =
+    useState<BCGeocoderAutocompleteData>();
+
+  // Initial routing options in the map controller
+  const initialRoutingOptions: RouteOptions = {
+    option: 'fastest',
+    roundTrip: false,
+    optimizeOrder: false,
+    ferrySchedule: false,
+    timeDependent: false,
+    traffic: false,
+    turnRestriction: false,
+    events: false,
+    crossingCost: false,
+    globalDistortionField: false,
+    turnCost: false,
+    localDistortionField: false,
+  };
+
+  const [routeOptions, setRouteOptions] = useState<RouteOptions>({
+    ...initialRoutingOptions,
+  });
+
+  /**
+   * Once Map is rendered in the dom this method must be called
+   * to set up map properties
+   */
+  const onRender = () => {
+    let m = L.map('map').setView([51.4, -116.9], 6);
+    // Saving map in state for future use
+    setMap(m);
+    addTileLayer(m);
+  };
+
+  /**
+   * Draws markers for all the locations
+   */
+  const drawAllMarkers = () => {
+    const mkrs: L.Marker[] = [];
+    mkrs.push(...drawStartingMarker());
+
+    if (selectedLocations?.length && map) {
+      selectedLocations.forEach((l) => {
+        if (l.latitude && l.longitude) {
+          let mkr = L.marker([+l.latitude, +l.longitude]);
+          mkr.bindTooltip(makeMarkerToolTip(l)).openTooltip();
+          mkrs.push(mkr);
+          mkr.addTo(map);
+        }
+      });
+    }
+    setMarkers(mkrs);
+    setMapBound(mkrs);
+  };
+
+  /**
+   * Appropriately zooms the map to show all markers
+   * @param mkrs
+   */
+  const setMapBound = (mkrs: L.Marker[]) => {
+    if (map && mkrs?.length) {
+      const group = new (L.featureGroup as any)(mkrs);
+      map.fitBounds(group.getBounds());
+    }
+  };
+
+  /**
+   * Draws a unique marker for the starting location
+   * @returns
+   */
+  const drawStartingMarker = (): L.Marker[] => {
+    if (startingLocation?.geometry?.coordinates && map) {
+      var startingIcon = new L.Icon({
+        iconUrl:
+          'https://raw.githubusercontent.com/pointhi/leaflet-color-markers/master/img/marker-icon-2x-red.png',
+        shadowUrl:
+          'https://cdnjs.cloudflare.com/ajax/libs/leaflet/0.7.7/images/marker-shadow.png',
+        iconSize: [25, 41],
+        iconAnchor: [12, 41],
+        shadowSize: [41, 41],
+      });
+
+      const [lng, lat] = startingLocation.geometry.coordinates;
+
+      const startingMarker = L.marker([lat, lng], { icon: startingIcon });
+      startingMarker.bindTooltip(
+        `<div style="text-align:center;"><b>Starting Location</b><br/>${safeGuardHtmlString(
+          startingLocation.properties.fullAddress
+        )}</div>`
+      );
+      startingMarker.addTo(map);
+      return [startingMarker];
+    }
+    return [];
+  };
+
+  /**
+   * Makes toolTip content from retail location markers
+   * @param l
+   * @returns
```

Comment:
Would be great to take a pass at the JSDoc to add some descriptions of the params/returns
Example:
```
@param l Business location to make marker for
@returns Marker tooltip as html
```